File: evidence/generated-systems/golang/inventory-on-golang/inventory-on-golang/internal/model/InventoryItem.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// InventoryItem Declaration
//==============================================================
type InventoryItem struct {
    gorm.Model
     QuantityOnHand                                                            string
    QuantityAvailable                                                            string
    QuantityReserved                                                            string
    UnitCost                                                            string
    LastUpdated                                                            time.Time
    SkuId         *uint
    Sku           *StockKeepingUnit `gorm:"foreignKey:SkuId"`
    WarehouseId         *uint
    Warehouse           *Warehouse `gorm:"foreignKey:WarehouseId"`
    LocationId         *uint
    Location           *StorageLocation `gorm:"foreignKey:LocationId"`
    LotId         *uint
    Lot           *Lot `gorm:"foreignKey:LotId"`
     SerialNumbers           []SerialNumber `gorm:"foreignKey:SerialNumbersFromInventoryItemId"`
     Transactions           []InventoryTransaction `gorm:"foreignKey:TransactionsFromInventoryItemId"`
     Reservations           []Reservation `gorm:"foreignKey:ReservationsFromInventoryItemId"`
    StockStatus                      StockStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/inventory-on-rails/inventoryonrails/app/models/inventoryitem.rb
class InventoryItem < ApplicationRecord
  enum StockStatus: [:Available, :Reserved, :Damaged, :Hold, :Quarantined, :InTransit, :PendingInspection]


  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  has_many :Sku, class_name: 'StockKeepingUnit'
  has_many :Warehouse, class_name: 'Warehouse'
  has_many :Location, class_name: 'StorageLocation'
  has_many :Lot, class_name: 'Lot'
  has_many :SerialNumbers, class_name: 'SerialNumber'
  has_many :Transactions, class_name: 'InventoryTransaction'
  has_many :Reservations, class_name: 'Reservation'

end
